Tests/Functional/PhpcrOdmRepositoryTest.php

```diff
@@ -12,68 +12,98 @@
 namespace Symfony\Cmf\Bundle\ResourceBundle\Tests\Functional;
 
 use Doctrine\ODM\PHPCR\Document\Generic;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Symfony\Cmf\Component\Resource\Repository\Resource\PhpcrOdmResource;
 
-class PhpcrOdmRepositoryTest extends PhpcrRepositoryTestCase
+class PhpcrOdmPhpcrRepositoryTest extends PhpcrRepositoryTestCase
 {
-    public function setUp()
-    {
-        $this->dm = $this->db('PHPCR')->getOm();
-
-        $this->db('PHPCR')->purgeRepository(true);
-        $this->db('PHPCR')->createTestNode();
-
-        $rootDocument =  $this->dm->find(null, '/test');
-        $document = new Generic();
-        $document->setNodeName('foo');
-        $document->setParentDocument($rootDocument);
-        $this->dm->persist($document);
-
-        $document = new Generic();
-        $document->setNodeName('bar');
-        $document->setParentDocument($rootDocument);
-        $this->dm->persist($document);
-        $this->dm->flush();
-
-        $this->repositoryRegistry = $this->container->get('cmf_resource.registry');
-    }
-
-    public function provideGet()
-    {
-        return array(
-            array('/foo', 'foo'),
-            array('/bar', 'bar'),
-            array('/', 'test'),
-        );
-    }
+    const REPOSITORY = 'test_repository_phpcr_odm';
 
     /**
      * @dataProvider provideGet
      */
     public function testRepositoryGet($path, $expectedName)
     {
-        $repository = $this->repositoryRegistry->get('test_repository');
+        $repository = $this->repositoryRegistry->get(self::REPOSITORY);
         $res = $repository->get($path);
         $this->assertNotNull($res);
         $document = $res->getPayload();
 
         $this->assertEquals($expectedName, $document->getNodeName());
     }
 
-    public function provideFind()
-    {
-        return array(
-            array('/*', 2),
-            array('/', 1),
-        );
-    }
-
     /**
      * @dataProvider provideFind
      */
     public function testRepositoryFind($pattern, $nbResults)
     {
-        $repository = $this->repositoryRegistry->get('test_repository');
+        $repository = $this->repositoryRegistry->get(self::REPOSITORY);
         $res = $repository->find($pattern);
         $this->assertCount($nbResults, $res);
     }
+
+    /**
+     * @dataProvider provideMove
+     */
+    public function testRepositoryMove($sourcePath, $targetPath, $expectedNodeName)
```

Comment:
These tests could go into the AbstractTestCase. Could you use raw PHPCR to assert for both the PHPCR-ODM and PHPCR?